flatdata-cpp/include/flatdata/internal/ResourceHandle.h

```diff
@@ -59,16 +66,18 @@ ResourceHandle::create( std::shared_ptr< std::ostream > stream )
         return std::unique_ptr< ResourceHandle >( );
     }
 
+    result->m_resource_reader = std::move( resource_reader );
+
     resource_storage::write_to_stream< resource_storage::size_type >( *result->m_stream, 0 );
     return result;
 }
 
-inline bool
+inline boost::optional< MemoryDescriptor >
```

Comment:
MemoryDescriptor itself has optional semantics, no need for optional (otherwise user has to double-check: !result || !*result)

Reply:
Makes sense. Resolved.